### PR TITLE
Fix not enough space error for keycloak container image quay.io/keycloak/keycloak:25.0

### PR DESCRIPTION
**Pull request by jcarranzan**
### Summary

In our daily jobs, we experienced `OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000006af400000, 4034920448, 0) failed; error=' Not enough space' (errno=12)#`.

It was some improvements with https://github.com/quarkus-qe/quarkus-test-framework/pull/1351 
Apart from that, the' HttpAdvancedIT' test also needs to be updated, and I did with this PR. 
I've tested in our Jenkins-job and the test PASSED.

Please select the relevant options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [ ] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by fedinskiy** [comment]
I believe we need a release of framework with this commit (https://github.com/quarkus-qe/quarkus-test-framework/pull/1351), don't we?

**Comment by jcarranzan** [comment]
> I believe we need a release of framework with this commit ([quarkus-qe/quarkus-test-framework#1351](https://github.com/quarkus-qe/quarkus-test-framework/pull/1351)), don't we?

Nope, because those changes already were bumped with the `1.6.0.Beta8` version.

File: http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java [hunk 1]
```diff
@@ -15,7 +15,8 @@
 public class HttpAdvancedIT extends BaseHttpAdvancedIT {
 
     @KeycloakContainer(command = { "start-dev", "--import-realm", "--hostname-strict=false" })
-    static KeycloakService keycloak = new KeycloakService(DEFAULT_REALM_FILE, DEFAULT_REALM, DEFAULT_REALM_BASE_PATH);
+    static KeycloakService keycloak = new KeycloakService(DEFAULT_REALM_FILE, DEFAULT_REALM, DEFAULT_REALM_BASE_PATH)
```
Comment (fedinskiy) [inline http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:18]: Do we need these for every test with keycloak? If not, what is so special about this one?

Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:18]: I don't know much about how Keycloak works, but yes these attributes are necessary in our test context with Keycloak configuration. I tried removing them keeping just the `DEFAULT_REALM_FILE` and the test failed, so `DEFAULT_REALM` and `DEFAULT_REALM_BASE_PATH` are also required for OIDC setup.

Comment (fedinskiy) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:18]: I was talking about `JAVA_OPTS` part on the next line

Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:18]: I've fixed the memory issue in our Jenkins job for this test which was the main issue.
 After analyzing other modules using KeycloakContainer and our KeycloakService, I found dependency differences that likely require less memory. For example, this specific module includes Netty dependencies:
```<dependency>
            <groupId>io.netty</groupId>
            <artifactId>netty-transport-native-epoll</artifactId>
            <classifier>linux-x86_64</classifier>
        </dependency>
        <dependency>
            <groupId>io.netty</groupId>
            <artifactId>netty-transport-native-kqueue</artifactId>
            <classifier>osx-x86_64</classifier>
        </dependency>
 ```

In this issue, with the netty version we use` 4.1.111.Final `: 
        https://github.com/netty/netty/issues/14221 - Memory leak in SSLHandler
      These dependencies may be consuming more memory on our Jenkins job machine...
      Look for example this netty issue:
      https://github.com/netty/netty/issues/8159 - Significant epoll native memory usage jump
      Not sure if related but also talk about epoll and use of much native memory allocated:
      https://github.com/netty/netty/issues/8288

Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:18]: But I've just realized that those dependencies are not related to the problematic test `HttpAdvancedIT` they are used for `DomainSocketIT`, I will investigate deeper, thanks @fedinskiy

Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:18]: It seems the test that increases memory consumption and leads to the docker Keycloak container fail is: `http2Server`
https://github.com/quarkus-qe/quarkus-test-suite/blob/c9f76aef2d947f4ec64b9bb9a7385ee3121ee558/http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/BaseHttpAdvancedIT.java#L128

Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:18]: I've tweaked the http2Server test with my last changes to fix the container issue.

File: http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/BaseHttpAdvancedIT.java [hunk 2]
```diff
@@ -244,7 +244,7 @@ private ResponsePredicateResult isHttp2x(HttpResponse<Void> resp) {
     private WebClientOptions defaultVertxHttpClientOptions() {
         return new WebClientOptions().setProtocolVersion(HttpVersion.HTTP_2).setSsl(true).setVerifyHost(false)
                 .setUseAlpn(true)
-                .setTrustStoreOptions(new JksOptions().setPassword(PASSWORD).setPath(defaultTruststore()));
+                .setTrustAll(true);
```
Comment (fedinskiy) [inline http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/BaseHttpAdvancedIT.java:247]: Why this? How does it solve the problem with keycloak memory?

File: http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java (hunk 1 above)
Comment (fedinskiy) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:18]: Do we still need `JAVA_OPTS` then?

Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java]: No, we don't, I've removed it, thanks.

File: http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/BaseHttpAdvancedIT.java (hunk 2 above)
Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/BaseHttpAdvancedIT.java:247]: Nope, it's not related, I made a mistake.

**Comment by jcarranzan** [comment]
None of those changes actually worked. I misinterpreted the results because, in the `HttpAdvancedIT` test, the JAVA_OPT property was set, which made the test pass. This led me to believe that certain changes had fixed the problem, so the only way to fix it is as I did at the beginning of this PR with `JAVA_OPT` property set.

PW: I could put also an annotation with the issue that you mentioned here: https://github.com/quarkusio/quarkus/issues/41813 WDYT?

File: http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java [hunk 3]
```diff
@@ -15,7 +15,8 @@
...
+            .withProperty("JAVA_OPTS", "-Xms512m -Xmx1g");
```
[the remark this answers is not in the document]
Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: 1. Yes it's the only test affected. The other ones are passing for instance `KeycloakWebappSecurityIT` or `HttpAdvancedReactiveIT`... I am currently investigating this by isolating different tests one by one in `BaseHttpAdvancedIT` to figure out which one is consuming the memory resources in the Keycloak container.

2. Ok restoring to the initial base code.

[the remark this answers is not in the document]
Comment (michalvavrik) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: > Yes it's the only test affected. 

That is strange. Any idea why?

[the remark this answers is not in the document]
Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: It seems this test is causing the issue:
https://github.com/quarkus-qe/quarkus-test-suite/blob/6de7fb98a148e02db062f79ff1e41238496df0b3/http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/BaseHttpAdvancedIT.java#L149

[the remark this answers is not in the document]
Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: It's weird, because I commented that test and Jenkins job passed, but according to the annotation that test is just executed for  `@EnabledOnQuarkusVersion(version = "1\\..*"`...and it's not the case...

[the remark this answers is not in the document]
Comment (michalvavrik) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: Right, it's some remnant that we forget and needs to be dropped, but it is highly unlikely it causes anything. Maybe Jenkins simply succeeded because it's flaky test.

[the remark this answers is not in the document]
Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: As I mentioned above that's the test root cause issue, I've recently tested again with @Disabled that test and PASSED in Jenkins job, so I will remove the test as @michalvavrik suggests.

[the remark this answers is not in the document]
Comment (michalvavrik) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java]: dropping this code makes sense, but I recommend to test that this fixes the issue on long series of Jenkins run

**Comment by jcarranzan** [comment]
Unfortunately even removing the test that "seems" the issue, it didn't solve the keycloak container error `Not enough space`...so it keeps being flaky and the only workaround keeps being my initial purpose with JAVA_OPTS.

File: http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (michalvavrik) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: > So, why is this the only test, which is affected?

+1

**Review by michalvavrik — COMMENTED** [review]
I have looked at the https://github.com/quarkus-qe/quarkus-test-framework/pull/1351 and https://github.com/quarkusio/quarkus/issues/43630 and I am not sure why do we need to set these parameters when there is `KeycloakContainer#memoryLimitMiB`. If I understand https://github.com/quarkusio/quarkus/pull/43601 correctly, Keycloak Dev Services does exactly same, just reduce container memory because Keycloak raise to that limit and no more.

But let's say we do, setting `JAVA_OPTS` to the  `-Xms512m -Xmx1g` means that heap can go up to 1g, but resident set size contains also metaspace and the metaspace can push you to OOM as well.

TL;DR; we have container memory limit set to 1000 MiB and you allow 1g for the heap, so what about the rest of RSS? My proposal is to test this well in many runs and link series of success runs with same configuration in the Slack. Thanks

Also, few questions:

- can we simply set `KeycloakContainer#memoryLimitMiB` to higher values for this test?
- any idea why it happens only for this test? I can imagine it is hard to tell the reason, but I found 6 other tests with identical Keycloak configuration just by simple fulltext search. And OOM happens on the Keycloak startup, which means it has nothing to do with any `@Test` method at all, netiher with application properties and Quarkus application either (because it happens before Quarkus is started).

This ^^ leads me to thinking that maybe there is some resource leak before `HttpAdvancedIT` is even started and by the time that Keycloak starts, there is already so little memory it cannot start.

But that you cannot investigate by guessing, you need more data, add there hooks and printout how much memory space there is on the executor machine disk before Keycloak is started.

**Comment by jcarranzan** [comment]
Thanks @michalvavrik ,you've  come to the same conclusion as me, I said this to Fedor yesterday on Slack: "Related to Keycloak container memory issues, I really think this has nothing to do with the tests." 
I will try to investigate it printing out the memory before keycloak starts, that's a good point.

**Comment by jcarranzan** [comment]
Ok printing the memory of the total memory and free memory before start the container seems the issue,  Jenkins logs : 
```
9:52:23 [INFO] Running io.quarkus.ts.http.advanced.HttpAdvancedIT
09:52:23 === System Memory Info ===
09:52:23 Total Physical Memory: 7695 MB
09:52:23 Free Physical Memory: 1330 MB
09:52:23 === Disk Space Info ===
09:52:23 Total Disk Space: 40847 MB
09:52:23 Free Disk Space: 32477 MB
09:52:24 07:52:24,522 INFO  ⭐  PKCS12 keystore and truststore generated successfully!
09:52:24 07:52:24,523 INFO  🔐  Key Store File: /tmp/quarkus-qe-certs3795654588981957403/quarkus-qe-keystore.p12
09:52:24 07:52:24,523 INFO  🔓  Trust Store File: /tmp/quarkus-qe-certs3795654588981957403/quarkus-qe-truststore.p12
09:52:24 07:52:24,563 INFO  [keycloak] Initialize service (quay.io/keycloak/keycloak:25.0)
09:52:24 07:52:24,590 Image pull policy will be performed by: DefaultPullPolicy()
09:52:24 07:52:24,593 Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
09:52:24 07:52:24,608 Testcontainers version: 1.20.2
09:52:25 07:52:25,223 Found Docker environment with Environment variables, system properties and defaults. Resolved dockerHost=unix:///run/user/600/podman/podman.sock
09:52:25 07:52:25,224 Docker host IP address is localhost
09:52:25 07:52:25,275 Connected to docker: 
09:52:25   Server Version: 4.9.4-rhel
09:52:25   API Version: 1.41
09:52:25   Operating System: rhel
09:52:25   Total Memory: 7695 MB
09:52:25 07:52:25,298 Pulling docker image: testcontainers/ryuk:0.9.0. Please be patient; this may take some time but only needs to be done once.
09:52:25 07:52:25,546 Starting to pull image
09:52:25 07:52:25,571 Pulling image layers:  2 pending,  1 downloaded,  0 extracted, (0 bytes/? MB)
09:52:25 07:52:25,577 Pulling image layers:  1 pending,  2 downloaded,  0 extracted, (0 bytes/? MB)
09:52:25 07:52:25,832 Pulling image layers:  0 pending,  3 downloaded,  0 extracted, (0 bytes/0 bytes)
09:52:25 07:52:25,835 Pulling image layers:  0 pending,  4 downloaded,  0 extracted, (0 bytes/0 bytes)
09:52:25 07:52:25,841 Pulling image layers:  0 pending,  4 downloaded,  0 extracted, (0 bytes/0 bytes)
09:52:25 07:52:25,842 Image testcontainers/ryuk:0.9.0 pull took PT0.543442853S
09:52:25 07:52:25,865 Creating container for image: testcontainers/ryuk:0.9.0
09:52:25 07:52:25,924 Container testcontainers/ryuk:0.9.0 is starting: e9b60ace3360744a689e7f627c6e1e3e0e388bd62904e4d84089d45d17f47d48
09:52:26 07:52:26,310 Container testcontainers/ryuk:0.9.0 started in PT0.44446503S
09:52:26 07:52:26,313 Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
09:52:26 07:52:26,313 Checking the system...
09:52:26 07:52:26,314 ✔︎ Docker server version should be at least 1.6.0
09:52:26 07:52:26,317 Pulling docker image: quay.io/keycloak/keycloak:25.0. Please be patient; this may take some time but only needs to be done once.
09:52:26 07:52:26,853 Starting to pull image
09:52:26 07:52:26,859 Pulling image layers:  0 pending,  1 downloaded,  0 extracted, (0 bytes/0 bytes)
09:52:27 07:52:27,361 Pulling image layers:  2 pending,  2 downloaded,  0 extracted, (0 bytes/? MB)
09:52:28 07:52:28,589 Pulling image layers:  1 pending,  3 downloaded,  0 extracted, (121 MB/? MB)
09:52:31 07:52:31,536 Pulling image layers:  0 pending,  4 downloaded,  0 extracted, (121 MB/217 MB)
09:52:31 07:52:31,539 Pulling image layers:  0 pending,  5 downloaded,  0 extracted, (121 MB/217 MB)
09:52:31 07:52:31,562 Pulling image layers:  0 pending,  5 downloaded,  0 extracted, (121 MB/217 MB)
09:52:31 07:52:31,562 Pull complete. 5 layers, pulled in 4s (downloaded 121 MB at 30 MB/s)
09:52:31 07:52:31,562 Image quay.io/keycloak/keycloak:25.0 pull took PT5.24515508S
09:52:31 07:52:31,569 Creating container for image: quay.io/keycloak/keycloak:25.0
09:52:31 07:52:31,740 Container quay.io/keycloak/keycloak:25.0 is starting: d86dffbe330cd6c624d131d533018e183f721ac21b3dfc469ce85b7c6fcca82d
09:57:32 07:57:32,044 Could not start container: java.lang.IllegalStateException: Wait strategy failed. Container exited with code 1
09:57:32 	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:533)
09:57:32 	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:354)
09:57:32 	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
09:57:32 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:344)
09:57:32 	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:330)
09:57:32 	at io.quarkus.test.services.containers.DockerContainerManagedResource.doStart(DockerContainerManagedResource.java:122)
09:57:32 	at io.quarkus.test.services.containers.DockerContainerManagedResource.start(DockerContainerManagedResource.java:72)
09:57:32 	at io.quarkus.test.bootstrap.BaseService.doStart(BaseService.java:356)
09:57:32 	at io.quarkus.test.bootstrap.BaseService.start(BaseService.java:245)
09:57:32 	at io.quarkus.test.bootstrap.QuarkusScenarioBootstrap.launchService(QuarkusScenarioBootstrap.java:172)
09:57:32 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
09:57:32 	at io.quarkus.test.bootstrap.QuarkusScenarioBootstrap.beforeAll(QuarkusScenarioBootstrap.java:73)
09:57:32 	at io.quarkus.test.bootstrap.QuarkusScenarioBootstrap.beforeAll(QuarkusScenarioBootstrap.java:50)
09:57:32 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$12(ClassBasedTestDescriptor.java:396)
09:57:32 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
09:57:32 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:396)
09:57:32 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:212)
09:57:32 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:85)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
09:57:32 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
09:57:32 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
09:57:32 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
09:57:32 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
09:57:32 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
09:57:32 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
09:57:32 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
09:57:32 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
09:57:32 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
09:57:32 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
09:57:32 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
09:57:32 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
09:57:32 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:198)
09:57:32 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:169)
09:57:32 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:93)
09:57:32 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:58)
09:57:32 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:141)
09:57:32 	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:57)
09:57:32 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:103)
09:57:32 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:85)
09:57:32 	at org.junit.platform.launcher.core.DelegatingLauncher.execute(DelegatingLauncher.java:47)
09:57:32 	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
09:57:32 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
09:57:32 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
09:57:32 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
09:57:32 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
09:57:32 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
09:57:32 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
09:57:32 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
09:57:32 Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*started in.*\s'
09:57:32 	at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)
09:57:32 	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)
09:57:32 	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:909)
09:57:32 	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:500)
09:57:32 	... 55 more
09:57:32 
09:57:32 07:57:32,060 Log output from the failed container:
09:57:32 OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000006af400000, 4034920448, 0) failed; error='Not enough space' (errno=12)
09:57:32 #
09:57:32 # There is insufficient memory for the Java Runtime Environment to continue.
09:57:32 # Native memory allocation (mmap) failed to map 4034920448 bytes. Error detail: committing reserved memory.
09:57:32 # An error report file with more information is saved as:
09:57:32 # /tmp/hs_err_pid84.log
09:57:32 
09:57:33 [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 309.3 s <<< FAILURE! -- in io.quarkus.ts.http.advanced.HttpAdvancedIT
09:57:33 [ERROR] io.quarkus.ts.http.advanced.HttpAdvancedIT -- Time elapsed: 309.3 s <<< ERROR!
09:57:33 org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:25.0
09:57:33 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:359)
09:57:33 	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:330)
09:57:33 	at io.quarkus.test.services.containers.DockerContainerManagedResource.doStart(DockerContainerManagedResource.java:122)
09:57:33 	at io.quarkus.test.services.containers.DockerContainerManagedResource.start(DockerContainerManagedResource.java:72)
09:57:33 	at io.quarkus.test.bootstrap.BaseService.doStart(BaseService.java:356)
09:57:33 	at io.quarkus.test.bootstrap.BaseService.start(BaseService.java:245)
09:57:33 	at io.quarkus.test.bootstrap.QuarkusScenarioBootstrap.launchService(QuarkusScenarioBootstrap.java:172)
09:57:33 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
09:57:33 	at io.quarkus.test.bootstrap.QuarkusScenarioBootstrap.beforeAll(QuarkusScenarioBootstrap.java:73)
09:57:33 	at io.quarkus.test.bootstrap.QuarkusScenarioBootstrap.beforeAll(QuarkusScenarioBootstrap.java:50)
09:57:33 	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
09:57:33 Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
09:57:33 	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
09:57:33 	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:344)
09:57:33 	... 10 more
09:57:33 Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
09:57:33 	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:563)
09:57:33 	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:354)
09:57:33 	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
09:57:33 	... 11 more
09:57:33 Caused by: java.lang.IllegalStateException: Wait strategy failed. Container exited with code 1
09:57:33 	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:533)
09:57:33 	... 13 more
09:57:33 Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*started in.*\s'
09:57:33 	at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)
09:57:33 	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)
09:57:33 	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:909)
09:57:33 	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:500)
09:57:33 	... 13 more
```

File: http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (fedinskiy) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: OK, this takes forever. @jcarranzan please:

1.  Create an issue in the TS to investigate why this tests (and only it ) fails
2. Add link to the issue to a TODO in this PR
And let's finally merge it.

**Review by fedinskiy — APPROVED** [review]

File: http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jcarranzan) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: ok, also I realized that fails in Podman Jenkins job machines, I will describe my insights in the issue.

**Comment by jcarranzan** [comment]
The issue opened in our TS:
https://github.com/quarkus-qe/quarkus-test-suite/issues/2106

File: http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (fedinskiy) [reply http/http-advanced/src/test/java/io/quarkus/ts/http/advanced/HttpAdvancedIT.java:19]: Do not forget to add the link to a todo comment here

**Review by fedinskiy — APPROVED** [review]
